src/main/java/org/kuali/kra/budget/document/BudgetDocumentRule.java

```diff
@@ -154,7 +154,86 @@ protected boolean processCustomSaveDocumentBusinessRules(Document document) {
     * @param budgetDocument
     * @return
     */
-    @SuppressWarnings("rawtypes")
+    @SuppressWarnings("rawtypes")  
+	protected boolean processBudgetProjectIncomeBusinessRule(BudgetDocument budgetDocument) {
+        boolean valid = true;
+        MessageMap errorMap = GlobalVariables.getMessageMap();
+        int i = 0;
```

Comment:
You can actually add this "i" into the for loop declaration and gain the increment for free.